proposals/array_interpolate_linear.json

```diff
@@ -0,0 +1,57 @@
+{
+    "id": "array_interpolate_linear",
+    "summary": "One-dimensional linear interpolation for arrays",
+    "description": "Performs a linear interpolation for each of the no-data values (`null`) in the array given, except for the first and last value.",
+    "categories": [
+        "arrays",
+        "math",
+        "math > interpolation"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "data",
+            "description": "An array of numbers and no-data values.\n\nIf the given array is a labeled array, the labels must have a natural/inherent label order and it expects the labels to be sorted accordingly. This is the default behavior in openEO for spatial and temporal dimensions.",
```

Comment:
> and it expects the labels to be sorted accordingly

Why is this necessary? It would be more user friendly to make the sorting  the responsibility of the backend implementation, I think

Reply:
I think there should be also explanation that not only the order of labels will be used, but also the value (if numerical).
e.g. that `interpolate({0: 10, 2: NaN, 10: 20})` results in `{0: 10, 2: 12, 10: 20}` and not `{0: 10, 2: 15, 10: 20}`

Reply:
> Why is this necessary? It would be more user friendly to make the sorting the responsibility of the backend implementation, I think

This is meant to be mostly a back-end responsibility. Spatial and temporal dimensions are sorted by default (temporal will likely be the most common use-case), all other dimensions are usually not usable for interpolation or must be sorted though. The reason is that the process on its own just from the array can't really know how to sort properly.

> I think there should be also explanation that not only the order of labels will be used, but also the value (if numerical).
> e.g. that `interpolate({0: 10, 2: NaN, 10: 20})` results in `{0: 10, 2: 12, 10: 20}` and not `{0: 10, 2: 15, 10: 20}`

Yes, I agree. It should be implicitly clear that the label values are used, but it can be made explicit. Unfortunately, I currently have no good idea how to describe that (without just providing an example).

Reply:
Maybe something like the folloowing?

> The linear interpolants are defined by the dimension labels (x) and the values in the array (y).

Reply:
See commit https://github.com/Open-EO/openeo-processes/pull/232/commits/7127bbe110b1faaf48c77c75020f178915a62768 for an attempt at solving your issues.

Reply:
> This is meant to be mostly a back-end responsibility

the "it expects the labels to be sorted accordingly" somewhat reads like it is responsibility of the user (because of the "expects"). Grammatical side note: what does the "it" refer to in that sentence?

I think if we can explain the x-y-interpolation details properly (for non-labeled and labeled arrays), we don't have to talk about label order in the first place.

Reply:
it = the process, will fix

Yes, it's the user responsibility, but I added the "This is the default behavior in openEO for spatial and temporal dimensions.", which was meant to make clear that for spatial and temporal dimensions usually no further action is required.